packages/compass/src/app/application.tsx

```diff
@@ -395,67 +324,170 @@ const app = {
+      previousVersion: this.previousVersion,
+      highestInstalledVersion: this.highestInstalledVersion,
+      version: this.version,
+    });
+  }
+
+  private setupZoomControls() {
+    const ZOOM_DEFAULT = 0;
+    const ZOOM_INCREMENT = 0.5;
+    const ZOOM_MAX = 5;
+    const ZOOM_MIN = -3;
+
+    const zoomReset = () => {
+      return webFrame.setZoomLevel(ZOOM_DEFAULT);
+    };
+    const zoomIn = () => {
+      const currentZoomLevel = webFrame.getZoomLevel();
+      const newZoomLevel = Math.min(
+        currentZoomLevel + ZOOM_INCREMENT,
+        ZOOM_MAX
+      );
+      return webFrame.setZoomLevel(newZoomLevel);
+    };
+    const zoomOut = () => {
+      const currentZoomLevel = webFrame.getZoomLevel();
+      const newZoomLevel = Math.max(
+        currentZoomLevel - ZOOM_INCREMENT,
+        ZOOM_MIN
+      );
+      return webFrame.setZoomLevel(newZoomLevel);
+    };
+
+    ipcRenderer?.on('window:zoom-reset', zoomReset);
+    ipcRenderer?.on('window:zoom-in', zoomIn);
+    ipcRenderer?.on('window:zoom-out', zoomOut);
+  }
+
+  private testUncaughtException() {
+    if (process.env.MONGODB_COMPASS_TEST_UNCAUGHT_EXCEPTION) {
+      queueMicrotask(() => {
+        throw new Error('fake exception');
+      });
+    }
+  }
+
+  private allowDevFeatureFlagsFromDevTools() {
+    // Lets us call `setShowDevFeatureFlags(true | false)` from DevTools.
+    (window as any).setShowDevFeatureFlags = async (
+      showDevFeatureFlags = true
+    ) => {
+      await defaultPreferencesInstance.savePreferences({ showDevFeatureFlags });
+    };
+  }
+
+  private preventDefaultBrowserBehaviorForDragAndDrop() {
+    // Drag and Drop Prevention
+    // Prevents default browser behavior for drag and drop events
+    // to avoid potential security issues
+    document.addEventListener('dragover', (evt) => evt.preventDefault());
+    document.addEventListener('drop', (evt) => evt.preventDefault());
+  }
+
+  private showCollectionSubMenu({ isReadOnly }: { isReadOnly: boolean }) {
+    void ipcRenderer?.call('window:show-collection-submenu', {
+      isReadOnly,
+    });
+  }
+
+  private hideCollectionSubMenu() {
+    void ipcRenderer?.call('window:hide-collection-submenu');
+  }
+
+  private showSettingsModal(tab?: SettingsTabId) {
+    globalAppRegistry?.emit('open-compass-settings', tab);
+  }
+
+  private async getWindowAutoConnectPreferences(): Promise<AutoConnectPreferences> {
+    return await ipcRenderer?.call(
+      'compass:get-window-auto-connect-preferences'
+    );
+  }
+
+  private async checkSecretStorageIsAvailable(): Promise<boolean> {
+    return await ipcRenderer?.call('compass:check-secret-storage-is-available');
+  }
+
+  async init() {
+    // Setup global error handling first to ensure all
+    // unhandled errors are properly logged and reported
+    this.setupGlobalErrorHandling();
+
+    // Inject CSP first to prevent any CSP violations.
+    injectCSP();
```

Comment:
I think it might be worth to still have this at the very top near "setup hadron" before any imports to make sure that csp applies to as much code that executes as possible